federation/pkg/kubefed/init/init_test.go

```diff
@@ -141,9 +166,11 @@ func TestInitFederation(t *testing.T) {
 				t.Errorf("[%d] expected error: %s, got: %s, output: %s", i, tc.expectedErr, cmdErrMsg, buf.String())
 			}
 		}
-
-		testKubeconfigUpdate(t, tc.federation, tc.lbIP, tc.kubeconfigGlobal, tc.kubeconfigExplicit)
 	}
+
+	//test config update as a separate test case
+	tc := testCases[0]
+	testKubeconfigUpdate(t, tc.federation, tc.lbIP, tc.kubeconfigGlobal, tc.kubeconfigExplicit)
```

Comment:
Why change this? Config update should be tested for each case, except the dry run case.